packages/ui/src/models/visualization/flows/abstract-camel-visual-entity.ts

```diff
@@ -163,6 +163,48 @@ export abstract class AbstractCamelVisualEntity<T extends object> implements Bas
     }
   }
 
+  isDefiniteNode(path?: string) {
+    if (path === 'route.from' || path === 'template.from') {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  switchSteps(options: { draggedNodePath: string; droppedNodePath?: string }) {
+    if (options.droppedNodePath === undefined) return;
+
+    const pathArray = options.droppedNodePath.split('.');
+    const last = pathArray[pathArray.length - 1];
+    const penultimate = pathArray[pathArray.length - 2];
+
+    const componentPath = options.draggedNodePath.split('.');
+    let stepsArray: ProcessorDefinition[];
+
+    if (!Number.isInteger(Number(last)) && Number.isInteger(Number(penultimate))) {
+      const componentModel = getValue(this.entityDef, componentPath?.slice(0, -1));
+      stepsArray = getArrayProperty(this.entityDef, pathArray.slice(0, -2).join('.'));
+
+      /** Remove the dragged node */
+      this.removeStep(options.draggedNodePath);
+
+      /** Add the dragged node before the drop target */
+      const desiredStartIndex = last === 'placeholder' ? 0 : Number(penultimate);
```

Comment:
I am unsure if there is any other possible way to check is this is a placeholder node here?
@lordrip

Reply:
not from the path itself :thinking:, if we talk about the IVizNode, then yes, you could check `vizNode.data.isPlaceholder`, but for the path only, is like you're doing right now.